gef.py

```diff
@@ -3528,11 +3528,30 @@ def process_lookup_path(name: str, perm: Permission = Permission.ALL) -> Optiona
         err("Process is not running")
         return None
 
+    matches = set()
+    ret = None
     for sect in gef.memory.maps:
-        if name in sect.path and sect.permission & perm:
-            return sect
+        try:
+            filename = pathlib.Path(sect.path).name
+        except:
+            filename = sect.path
 
-    return None
+        if name in filename and sect.permission & perm:
+            if ret is None:
+                ret = sect
+            matches.add(sect.path)
+
+    matches_count = len(matches)
+
+    if matches_count > 1:
+        warn(f"{matches_count} matches! You should probably refine your search!")
+
+        for x in matches:
+            warn(f"- '{x}'")
+
+        warn("Returning the first match")
+
+    return ret
```

Comment:
This can be simplified because `ret` is not necessary:
 - if `len(matches)` is zero return `None` immediately
 - if strictly more than 1, this warning is fine
 - in any case return `matches[0]`

Reply:
I can do this, but this means we have to change matches to be a set of `Section` instead of a set of `str`. Sections will always be different, even if the path is the same, so we can even change it to a list of sections, but we then lose the advantage of having a set of paths to compute quickly `len(matches)`.
The problem is that we then have to compute `len(set(x.path for x in matches))`, which means iterating again through (a subset of) the sections to get what is stored in matches as a set of str in the current code, which doesn't seem cleaner to me.
If you see a cleaner way to do it I would be happy to implement this (or if you still think that just removing the `ret` variable and storing the sections instead of the paths in `matches` is better, I can make the changes)